analyses/sample-distribution-analysis/01-filter-across-types.R

```diff
@@ -0,0 +1,143 @@
+# This script filters the given dataset to produce a summarized visualization
+# of key variables within the dataset.
+#
+# Chante Bethell for CCDL 2019
+
+# magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+# Function to filter based on primary_site 
+location_fn <- function(location) {
+  disease_type_vector <- brain_location %>%
+    dplyr::arrange(dplyr::desc(n)) %>%
+    dplyr::filter(stringr::str_detect(primary_site, location)) %>%
+    dplyr::pull(disease_type_new)
+  unique(disease_type_vector)
+}
+
+# Create directories to hold the output.
+if (!dir.exists("results")) {
+  dir.create("results")
+}
+if (!dir.exists("plots")) {
+  dir.create("plots")
+}
+
+# Read in dataset
+df2 <- data.frame(readr::read_tsv(
+  file.path("data", "pbta-histologies.tsv")
+))
+
+# Remove na's
+df2 <- df2 %>%
+  dplyr::filter(!is.na(disease_type_new))
+
+# data.frame with the count of each unique cancer type expression
+disease_expression <- df2 %>%
+  dplyr::group_by(disease_type_new) %>%
+  dplyr::count(name = "count") %>%
+  dplyr::arrange(dplyr::desc(count))
+
+# Calculate the total count of the dataset
+sum_count <- sum(disease_expression$count)
+
+# Create a percent variable and round to 4 decimal places 
+# (so values will have 2 decimal places as percentages) 
+disease_expression <- disease_expression %>%
+  dplyr::mutate(percent = paste0((round(count / sum_count, 4) * 100), "%"))
+
+# Reorder the columns to be displayed in descending order by count on the plot
+disease_expression$disease_type_new <- with(disease_expression,
+                                            reorder(disease_type_new, -count))
+
+# Write to tsv file
+readr::write_tsv(disease_expression,
+                 file.path("results",
+                           "disease_expression.tsv"))
+
+# Create a bar plot of sample distribution across cancer types
+gg_types <- disease_expression %>%
+  ggplot2::ggplot(ggplot2::aes(x = disease_type_new, y = count, fill = count)) +
+  ggplot2::geom_col() +
+  ggplot2::theme_bw() +
+  ggplot2::labs(x = "Cancer Types", y = "Count",
+                title = "Sample Distribution Across Cancer Types") +
+  ggplot2::scale_y_continuous(breaks = seq(0, 500, by = 100)) +
+  ggplot2::theme(axis.text.x = element_text(
```

Comment:
```suggestion
  ggplot2::theme(axis.text.x = ggplot2::element_text(
```
Based on the test failure